llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -658,6 +658,32 @@ static InstructionsState getSameOpcode(ArrayRef<Value *> VL,
   unsigned AltOpcode = Opcode;
   unsigned AltIndex = BaseIndex;
 
+  bool SwappedPredsCompatible = [&]() {
+    if (!IsCmpOp)
+      return false;
+    SetVector<unsigned> UniquePreds, UniqueNonSwappedPreds;
+    UniquePreds.insert(BasePred);
+    UniqueNonSwappedPreds.insert(BasePred);
+    for (Value *V : VL) {
+      auto *I = dyn_cast<CmpInst>(V);
+      if (!I) {
+        UniquePreds.clear();
+        UniqueNonSwappedPreds.clear();
+        break;
```

Comment:
why not just return false?

Reply:
Yep, remains from the initial version of the patch, will fix.